### PR TITLE
feat(router): add domain types, admin core changes and other prerequisites for 3ds external authentication flow

### PR DESCRIPTION
**Pull request by sai-harsha-vardhan**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
add domain types, admin core changes and other prerequisites for 3ds external authentication flow

### Additional Changes

- [ ] This PR modifies the API contract
- [x] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Compiler guided and sanity testing of payments

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/authentication/types.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+pub struct AuthenticationData {
+    pub maximum_supported_version: (i64, i64, i64),
+    pub threeds_server_transaction_id: String,
+    pub cavv: Option<String>,
+    pub authn_flow_type: Option<AuthNFlowType>,
+    pub three_ds_method_data: ThreeDsMethodData,
+    pub message_version: String,
+    pub eci: Option<String>,
+    pub trans_status: api_models::payments::TransactionStatus,
+    pub acquirer_details: Option<AcquirerDetails>,
```
Comment (Narayanbhat166) [inline crates/router/src/core/authentication/types.rs:41]: few of these fields can be stored in the table, ones which are to be updated.

File: crates/router/src/types/api/authentication.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+pub struct PostAuthentication;
+use crate::{services, types};
+
+#[derive(Clone, serde::Deserialize, Debug, serde::Serialize)]
+pub struct AcquirerDetails {
+    pub acquirer_bin: String,
+    pub acquirer_merchant_mid: String,
+}
+
+#[derive(Clone, serde::Deserialize, Debug, serde::Serialize)]
```
Comment (Narayanbhat166) [inline crates/router/src/types/api/authentication.rs:21]: core types need not have serialize and Deserialize

**Review by Narayanbhat166 — APPROVED** [review]

**Review by prajjwalkumar17 — APPROVED** [review]

File: crates/router/src/types/api/authentication.rs (hunk 2 above)
Comment (sai-harsha-vardhan) [reply crates/router/src/types/api/authentication.rs:21]: I'll take this up in next pr

**Review by ThisIsMani — APPROVED** [review]
Dashboard specific changes looks good.

**Pull request merged**